### PR TITLE
Add ability to search for updates with different artifact id and / or group id

### PR DESCRIPTION
**Pull request by scottrice10**
This builds from the [custom group migrations](https://github.com/scala-steward-org/scala-steward/pull/1298) work. The author there, @kubukoz, mentioned wanting to "tackle changing the artifact name... in another PR", and this is my attempt to do that.

I do not know of open source Scala projects that could be used to test upgrading with a different artifact id, or updating both the group id and artifact id. Ideally, someone in the community would know of such examples so that they can be added to tests here.

At my company, we are using scala-steward and need to be able to update a privately hosted library like this:
```
"com.company" %% "old-artifact-name" % "1.0.0" ->  "com.company" %% "new-artifact-name" % "2.0.0"
```
so that is my primary purpose for this PR.

I opted to rename the cli argument from `--group-migrations` to `--artifact-migrations`, and keep the logic for updating either / both of the artifacts in one place, since I think that was simpler and easier to handle the "both" case. However, this is a backwards incompatible change, so if this is undesired, I could break the artifact id rename into a separate cli argument and logic in code.

### CODE REVIEW

File: docs/artifact-migrations.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+In this example, scala-steward will look in the project for dependencies with the before group id, "com.geirsson", and 
+artifact id "sbt-scalafmt". If found, scala-steward will search for updates using the after group id, "org.scalameta", 
+artifact id "sbt-scalafmt", and version greater than the initial version, 2.0.0. If found, scala-steward will update
+to the after group id and latest version.
+
+The fields `groupIdBefore` and `artifactIdBefore` are optional. If just `groupIdBefore` is specified, as in the previous
+example, then only the group id will get renamed. If just `artifactIdBefore` is specified, then only the artifact id
+will get renamed. Specifying both `groupIdBefore` and `artifactIdBefore` will rename both.
+
+[migrations]: https://github.com/fthomas/scala-steward/blob/master/modules/core/src/main/resources/artifact-migrations.conf
```
Comment (fthomas) [inline docs/artifact-migrations.md:31]:
```suggestion
[migrations]: https://github.com/scala-steward-org/scala-steward/blob/master/modules/core/src/main/resources/artifact-migrations.conf
```

File: docs/running.md [hunk 2]
```diff
@@ -196,3 +195,7 @@ echo "${SCALA_STEWARD_TOKEN}"
 ```
 7. add the `repos.md` file 
 8. (*optional*) create a new schedule to trigger the pipeline on a daily/weekly basis
+
+
+[scalafixmigrations]: https://github.com/fthomas/scala-steward/blob/master/docs/scalafix-migrations.md
+[artifactmigrations]: https://github.com/fthomas/scala-steward/blob/master/docs/artifact-migrations.md
```
Comment (fthomas) [inline docs/running.md:201]:
```suggestion
[scalafixmigrations]: https://github.com/scala-steward-org/scala-steward/blob/master/docs/scalafix-migrations.md
[artifactmigrations]: https://github.com/scala-steward-org/scala-steward/blob/master/docs/artifact-migrations.md
```

File: modules/core/src/test/scala/org/scalasteward/core/mock/MockContext.scala [hunk 3]
```diff
@@ -75,8 +75,8 @@ object MockContext {
   implicit val filterAlg: FilterAlg[MockEff] = new FilterAlg[MockEff]
   implicit val versionsCache: VersionsCache[MockEff] =
     new VersionsCache[MockEff](config.cacheTtl, new JsonKeyValueStore("versions", "1"))
-  implicit val groupMigrations: GroupMigrations =
-    GroupMigrations.create[MockEff].runA(MockState.empty).unsafeRunSync()
+  implicit val groupMigrations: ArtifactMigrations =
```
Comment (fthomas) [inline modules/core/src/test/scala/org/scalasteward/core/mock/MockContext.scala:78]:
```suggestion
  implicit val artifactMigrations: ArtifactMigrations =
```

File: modules/core/src/main/scala/org/scalasteward/core/application/Context.scala [hunk 4]
```diff
@@ -58,7 +58,7 @@ object Context {
       implicit0(fileAlg: FileAlg[F]) = FileAlg.create[F]
       implicit0(migrationAlg: MigrationAlg) <-
         Resource.liftF(new MigrationsLoader[F].loadAll(config.scalafixCfg).map(new MigrationAlg(_)))
-      implicit0(groupMigration: GroupMigrations) <- Resource.liftF(GroupMigrations.create[F])
+      implicit0(groupMigration: ArtifactMigrations) <- Resource.liftF(ArtifactMigrations.create[F])
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/application/Context.scala:61]:
```suggestion
      implicit0(artifactMigrations: ArtifactMigrations) <- Resource.liftF(ArtifactMigrations.create[F])
```

File: modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala [hunk 5]
```diff
@@ -27,7 +27,7 @@ import scala.concurrent.duration.FiniteDuration
 final class UpdateAlg[F[_]](implicit
     filterAlg: FilterAlg[F],
     versionsCache: VersionsCache[F],
-    groupMigrations: GroupMigrations,
+    groupMigrations: ArtifactMigrations,
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala:30]:
```suggestion
    artifactMigrations: ArtifactMigrations,
```

File: modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala [hunk 6]
```diff
@@ -40,7 +40,7 @@ final class UpdateAlg[F[_]](implicit
       maybeNewerVersions = Nel.fromList(versions.filter(_ > current))
       maybeUpdate = maybeNewerVersions
         .map(vs => Update.Single(CrossDependency(dependency.value), vs.map(_.value)))
-        .orElse(groupMigrations.findUpdateWithNewerGroupId(dependency.value))
+        .orElse(groupMigrations.findUpdateWithRenamedArtifact(dependency.value))
```
Comment (fthomas) [inline modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala:43]:
```suggestion
        .orElse(artifactMigrations.findUpdateWithRenamedArtifact(dependency.value))
```

**Review by fthomas — COMMENTED** [review]
This looks good to me, thanks @scottrice10! I agree with you that having both migrations in the same place is nicer than to having them separated.

I added a couple of suggested changes to rename the old `groupMigrations` name to `artifactMigrations`.

File: docs/artifact-migrations.md (hunk 1 above)
Comment (scottrice10) [reply docs/artifact-migrations.md]: I did a Find and Repace from https://github.com/fthomas/scala-steward -> https://github.com/scala-steward-org/scala-steward in all places.

File: modules/core/src/test/scala/org/scalasteward/core/mock/MockContext.scala (hunk 3 above)
Comment (scottrice10) [reply modules/core/src/test/scala/org/scalasteward/core/mock/MockContext.scala]: Updated as suggested.

File: modules/core/src/main/scala/org/scalasteward/core/application/Context.scala (hunk 4 above)
Comment (scottrice10) [reply modules/core/src/main/scala/org/scalasteward/core/application/Context.scala]: Updated as suggested.

File: modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala (hunk 5 above)
Comment (scottrice10) [reply modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala]: Updated as suggested.

File: modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala (hunk 6 above)
Comment (scottrice10) [reply modules/core/src/main/scala/org/scalasteward/core/update/UpdateAlg.scala]: Updated as suggested.

**Review by fthomas — APPROVED** [review]

**Pull request merged**